packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1014,6 +1020,45 @@ describe("BnsConnection", () => {
 
       connection.disconnect();
     });
+
+    it("reports DeliverTx errors for search by ID", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = connection.chainId();
+
+      const { profile, mainWalletId } = await userProfileWithFaucet(chainId);
+      // this will never have tokens, but can try to sign
+      const brokeIdentity = await profile.createIdentity(mainWalletId, chainId, HdPaths.simpleAddress(1234));
+
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        creator: brokeIdentity,
+        recipient: await randomBnsAddress(),
+        memo: "Sending from empty",
+        amount: defaultAmount,
+      };
+
+      const nonce = await connection.getNonce({ pubkey: brokeIdentity.pubkey });
+      const signed = await profile.signTransaction(mainWalletId, brokeIdentity, sendTx, bnsCodec, nonce);
+      const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+      const transactionIdToSearch = response.transactionId;
+      await response.blockInfo.waitFor(info => info.state !== TransactionState.Pending);
+
+      await tendermintSearchIndexUpdated();
```

Comment:
This is an interesting issue.
We use sleep now.
Digging into those issues with bugs in wallet-demo, I found the dispatch code. Maybe we can get a tendermint event for this case (`search_index_updated`) shortly after all headers and tx events are emitted.

Reply:
(Anton, tendermint core dev who build the indexing system, also proposed this `IndexedBlock` event... so maybe in the not too distance future)

Reply:
Right now we consistently delay 50ms between block event and searches. This is done in all Tendermint and BNS tests. Before we had many race condition issues and tests that ran on a dev machine but occasionally failed on Travis. Since we use the 50 ms, everything is fine.

But of course, having an event that fires after the new block is indexed would be much more convenient.